ENIGMAsystem/SHELL/Widget_Systems/Win32/dialogs.cpp

```diff
@@ -97,29 +126,73 @@ static INT_PTR CALLBACK ShowInfoProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPA
   return DefWindowProc(hwndDlg, uMsg, wParam, lParam);
 }
 
-static INT_PTR CALLBACK GetStrProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
+WCHAR wstrPromptStr[4096];
+WCHAR wstrTextEntry[MAX_PATH];
+bool HideInput = 0;
+
+void ClientResize(HWND hWnd, int nWidth, int nHeight)
 {
-  if (uMsg == WM_INITDIALOG) {
-    SetWindowText(hwndDlg, gs_cap.c_str());
-    SetDlgItemText(hwndDlg, 12, gs_def.c_str());
-    SetDlgItemText(hwndDlg, 13, gs_message.c_str());
-  }
+  RECT rcClient, rcWind;
+  POINT ptDiff;
+  GetClientRect(hWnd, &rcClient);
+  GetWindowRect(hWnd, &rcWind);
+  ptDiff.x = (rcWind.right - rcWind.left) - rcClient.right;
+  ptDiff.y = (rcWind.bottom - rcWind.top) - rcClient.bottom;
+  MoveWindow(hWnd, rcWind.left, rcWind.top, nWidth + ptDiff.x, nHeight + ptDiff.y, TRUE);
+}
 
-  if (uMsg == WM_COMMAND) {
-    if (wParam == 2 || wParam == 11) {
-      gs_str_submitted = "";
-      gs_form_canceled = 1;
-      EndDialog(hwndDlg, 1);
-    } else if (wParam == 10) {
-      char strget[1024];
-      GetDlgItemText(hwndDlg, 12, strget, 1024);
-      gs_str_submitted = strget;
+LRESULT CALLBACK InputBoxHookProc(HWND hWndDlg, UINT Msg, WPARAM wParam, LPARAM lParam)
```

Comment:
That hook procedure is used for multiple functions (4 of them to be exact) that's why I decided to have a more general name that describes all of them equally correct rather than naming it after one specific function which wouldn't make too much sense for the other 3 that use the procedure. But I'm fine with changing its name back if that'll be the only way this bad boy's getting merged. :P

Reply:
Alright, you got me on that one, that's not a deal breaker then, I'll let this one slide.